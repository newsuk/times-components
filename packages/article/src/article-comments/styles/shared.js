import styleguide from "@times-components/styleguide";

const { colours, fontFactory, spacing } = styleguide();
const styles = {
  container: {
    alignItems: "center",
    borderTopColor: colours.functional.keyline,
    borderTopWidth: 1,
    borderStyle: "solid",
    marginBottom: spacing(5),
    width: "100%"
  },
  headline: {
    color: colours.functional.primary,
    ...fontFactory({
      font: "headline",
      fontSize: "commentsHeadline"
    }),
    maxWidth: 315,
    paddingBottom: spacing(2),
    paddingTop: spacing(6),
    textAlign: "center"
  },
  supporting: {
    color: colours.functional.secondary,
    ...fontFactory({
      font: "supporting",
      fontSize: "tertiary"
    }),
    textAlign: "center",
    maxWidth: 280
  },
  link: {
    color: colours.functional.action
  },
  button: {
    marginTop: spacing(5),
    maxWidth: 215,
    marginBottom: spacing(1)
  }
};

export default styles;
