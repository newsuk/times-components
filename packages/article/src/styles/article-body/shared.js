import styleguide from "@times-components/styleguide";

const sharedStyles = scale => {
  const { colours, fontFactory, spacing } = styleguide({ scale });
  return {
    leadAsset: {
      marginBottom: spacing(2)
    },
    articleTextElement: {
      ...fontFactory({
        font: "body",
        fontSize: "bodyMobile"
      }),
      marginBottom: spacing(5),
      color: colours.functional.primary
    },
    articleMainContentRow: {
      paddingLeft: spacing(2),
      paddingRight: spacing(2)
    },
    articleLink: {
      color: colours.functional.action,
      ...fontFactory({
        font: "body",
        fontSize: "bodyMobile"
      }),
      marginBottom: spacing(5),
      marginTop: 0
    },
    ad: {
      borderTopColor: colours.functional.keyline,
      borderBottomColor: colours.functional.keyline,
      borderTopWidth: 1,
      borderBottomWidth: 1,
      paddingTop: spacing(2),
      paddingBottom: spacing(2),
      marginBottom: spacing(4)
    },
    interactiveContainer: {
      paddingRight: spacing(2),
      paddingLeft: spacing(2),
      marginBottom: spacing(4)
    },
    primaryContainer: {
      width: "100%",
      flexDirection: "column",
      paddingBottom: spacing(5)
    }
  };
};

export default sharedStyles;
