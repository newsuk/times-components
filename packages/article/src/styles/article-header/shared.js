import styleguide from "@times-components/styleguide";

const { colours, fontFactory, spacing } = styleguide();
const sharedStyles = {
  articleFlag: {
    flexDirection: "row",
    marginBottom: spacing(2)
  },
  articleFlagContainer: {
    marginRight: spacing(3)
  },
  articleHeadLineText: {
    ...fontFactory({
      font: "headline",
      fontSize: "headline"
    }),
    color: colours.functional.brandColour,
    marginBottom: 7
  },
  articleLabel: {
    paddingBottom: spacing(1),
    paddingTop: spacing(1)
  },
  standFirst: {
    ...fontFactory({
      font: "headlineRegular",
      fontSize: "smallestHeadline"
    }),
    color: colours.functional.primary,
    paddingBottom: spacing(2)
  }
};

export default sharedStyles;
