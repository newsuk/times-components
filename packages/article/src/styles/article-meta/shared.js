import styleguide from "@times-components/styleguide";

const { colours, fontFactory, spacing } = styleguide();
const sharedStyles = {
  articleMeta: {
    borderBottomColor: colours.functional.keyline,
    borderBottomWidth: 1,
    marginBottom: spacing(4),
    marginLeft: spacing(2),
    marginRight: spacing(2),
    paddingLeft: 0,
    paddingRight: 0
  },
  articleMetaElement: {
    borderTopColor: colours.functional.keyline,
    borderTopWidth: 1
  },
  articleMiddleContainer: {
    paddingTop: 9
  },
  byline: {
    ...fontFactory({
      font: "supporting",
      fontSize: "cardMeta"
    }),
    color: colours.functional.secondary
  },
  datePublication: {
    color: colours.functional.secondary,
    ...fontFactory({
      font: "supporting",
      fontSize: "cardMeta"
    })
  }
};

export default sharedStyles;
