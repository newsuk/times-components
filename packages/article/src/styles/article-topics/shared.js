import { colours } from "@times-components/styleguide";

const sharedStyles = {
  topicsContainer: {
    borderTopColor: colours.functional.keyline,
    borderTopWidth: 1
  },
  topicsMetaContainer: {
    justifyContent: "flex-start"
  }
};

export default sharedStyles;
