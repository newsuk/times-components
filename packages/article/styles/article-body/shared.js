import { fonts, colours, spacing } from "@times-components/styleguide";

const sharedStyles = {
  leadAsset: {
    marginBottom: spacing.default
  },
  articleTextElement: {
    fontFamily: fonts.body,
    lineHeight: 26,
    fontSize: 17,
    marginBottom: spacing.l,
    color: colours.functional.primary
  },
  articleMainContentRow: {
    paddingLeft: spacing.default,
    paddingRight: spacing.default
  }
};

export default sharedStyles;
