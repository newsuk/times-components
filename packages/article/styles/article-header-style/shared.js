import { StyleSheet } from "react-native";
import {
  COLOUR_HEADLINE,
  COLOUR_TEXT,
  FONT_FAMILY_STAND_FIRST
} from "../const";

const sharedStyles = StyleSheet.create({
  articleHeadLineText: {
    fontSize: 30,
    lineHeight: 32,
    color: COLOUR_HEADLINE,
    marginBottom: 7
  },
  standFirst: {
    fontSize: 21,
    lineHeight: 26,
    fontFamily: FONT_FAMILY_STAND_FIRST,
    color: COLOUR_TEXT,
    paddingBottom: 9
  }
});

export default sharedStyles;
