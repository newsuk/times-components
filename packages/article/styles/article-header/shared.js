import { fonts } from "@times-components/styleguide";

const sharedStyles = {
  articleHeadLineText: {
    fontSize: 30,
    lineHeight: 32,
    color: "#1d1d1b",
    marginBottom: 7,
    fontFamily: fonts.headline
  },
  standFirst: {
    fontSize: 20,
    lineHeight: 26,
    fontFamily: fonts.headlineRegular,
    color: "#333333",
    paddingBottom: 9
  },
  articleLabel: {
    paddingTop: 4,
    paddingBottom: 5
  },
  articleFlag: {
    marginBottom: 11,
    flexDirection: "row"
  },
  articleFlagContainer: {
    marginRight: 14
  }
};

export default sharedStyles;
