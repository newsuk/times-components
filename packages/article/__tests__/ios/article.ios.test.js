import shared from "../shared";

describe("Article test on ios", () => {
  shared();
});
