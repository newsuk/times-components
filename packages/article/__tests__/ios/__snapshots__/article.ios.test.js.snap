// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`1. an error 1`] = `
<View>
  <View>
    <Image
      accessibilityLabel="Server error"
      resizeMode="contain"
      source={
        Object {
          "testUri": "../../../packages/article/assets/article-error.png",
        }
      }
    />
    <Text>
      Something's gone wrong
    </Text>
    <Text>
      We can't load the page you have requested. Please check your network connection and retry to continue
    </Text>
  </View>
  <Button
    title="Retry"
  />
</View>
`;

exports[`2. a full article with an image as the lead asset 1`] = `
<RCTScrollView
  initialNumToRender={10}
  maxToRenderPerBatch={10}
  numColumns={1}
  onEndReachedThreshold={2}
  scrollEventThrottle={50}
  updateCellsBatchingPeriod={50}
  windowSize={21}
>
  <View>
    <View>
      <View>
        <ModalImage
          aspectRatio={1.7777777777777777}
          caption="Some Caption"
          credits="Some Credits"
          highResSize={1440}
          uri="https://crop169.io"
        />
      </View>
    </View>
    <View>
      <View>
        <View>
          <ArticleLabel
            color="#333333"
            title="Some Label"
          />
        </View>
        <Text
          selectable={true}
        >
          Some Headline
        </Text>
        <Text
          accessibilityRole="heading"
          aria-level="2"
        >
          Some Standfirst
        </Text>
        <View>
          <View>
            <NewArticleFlag />
          </View>
        </View>
      </View>
    </View>
    <View>
      <View>
        <View>
          <Text>
            <ArticleBylineWithLinks
              ast={
                Array [
                  Object {
                    "attributes": Object {},
                    "children": Array [
                      Object {
                        "attributes": Object {
                          "value": "Some byline",
                        },
                        "children": Array [],
                        "name": "text",
                      },
                    ],
                    "name": "inline",
                  },
                ]
              }
              color="#333333"
            />
          </Text>
        </View>
        <View>
          <Text>
            Friday March 13 2015, 6:54pm, The Times
          </Text>
        </View>
      </View>
    </View>
    <View>
      <View>
        <ArticleImage
          captionOptions={
            Object {
              "caption": "An image caption",
              "credits": "The image credits",
            }
          }
          imageOptions={
            Object {
              "display": "primary",
              "ratio": "1500:1000",
              "uri": "https://image.io",
            }
          }
        />
      </View>
    </View>
    <View>
      <Text
        accessibilityRole="link"
      >
        Some Link
      </Text>
    </View>
    <View>
      <View>
        <Text
          selectable={true}
        >
          Some content
        </Text>
      </View>
    </View>
    <View>
      <View>
        <PullQuote
          caption="AName"
          content="A pull quote"
          twitter="@AName"
        />
      </View>
    </View>
    <View>
      <View>
        <Video
          accountId="57838016001"
          height={421.875}
          policyKey="1.2.3.4"
          poster={
            Object {
              "uri": "https://image.io",
            }
          }
          skySports={false}
          videoId="4084164751001"
          width={750}
        />
        <View>
          <View>
            <Text>
              This is video caption
            </Text>
          </View>
        </View>
      </View>
    </View>
    <View>
      <View>
        <ArticleTopics
          topics={
            Array [
              Object {
                "name": "Topic",
                "slug": "topic",
              },
            ]
          }
        />
      </View>
    </View>
    <View>
      <RelatedArticles
        slice={
          Object {
            "__typename": "StandardSlice",
            "items": Array [
              Object {
                "article": Object {
                  "__typename": "Article",
                  "byline": Array [],
                  "hasVideo": false,
                  "headline": "RA Headline",
                  "id": "ra-1",
                  "label": "RA Label",
                  "leadAsset": Object {
                    "__typename": "Image",
                    "crop169": Object {
                      "__typename": "Crop",
                      "url": "https://racrop169.io",
                    },
                    "crop32": Object {
                      "__typename": "Crop",
                      "url": "https://racrop32.io",
                    },
                    "title": "RA Title",
                  },
                  "publicationName": "TIMES",
                  "publishedTime": "2015-03-23T19:39:39.000Z",
                  "shortHeadline": "Headline",
                  "shortIdentifier": "2k629tpvh",
                  "slug": "this-is-slug",
                  "summary105": Array [],
                  "summary125": Array [],
                  "summary145": Array [],
                  "summary160": Array [],
                  "summary175": Array [],
                  "summary225": Array [],
                  "url": "https://ra.io",
                },
              },
            ],
            "sliceName": "StandardSlice",
          }
        }
      />
    </View>
    <View />
  </View>
</RCTScrollView>
`;

exports[`2. an empty article 1`] = `
<RCTScrollView
  initialNumToRender={10}
  maxToRenderPerBatch={10}
  numColumns={1}
  onEndReachedThreshold={2}
  scrollEventThrottle={50}
  updateCellsBatchingPeriod={50}
  windowSize={21}
>
  <View>
    <View>
      <View>
        <Text>
          NEW BREXIT REFERENDUM
        </Text>
        <View>
          <View>
            <View />
            <Text>
              Another ministerial aide, 
              <Text
                accessibilityRole="link"
              >
                Robert Courts
              </Text>
              , resigned saying that he could not look himself in the mirror if he backed the Chequers proposals.
            </Text>
          </View>
          <View>
            <View />
            <Text>
              Downing Street and Conservative Party bosses were seeking to assess the scale of a grassroots rebellion.
            </Text>
          </View>
          <View>
            <View />
            <Text>
              Dennis Mitchell’s high levels of testosterone in 1998 were caused by having sex with his wife "at least four times" on her birthday, according to the sprinter. USA Track and Field believed him, the IAAF did not
            </Text>
          </View>
          <View>
            <View />
            <Text>
              Chuka Umunna, leader of the 
              <Text>
                Labour Remainers
              </Text>
              , dismissed Mrs May’s 
              <Text>
                compromise
              </Text>
               as “anti-worker and anti-business” and said that it would “royally screw” Britain’s service sector.
            </Text>
          </View>
        </View>
      </View>
    </View>
    <View>
      <View>
        <Text
          selectable={true}
        >
          Mr Gove has previously said that memorising a poem is to “own a great work of art forever”. Andrew Motion, the former Poet Laureate, wrote in the 
          <Text
            accessibilityRole="link"
          >
            police disclosure of evidence
          </Text>
          <Text>
             Telegraph 
          </Text>
          last year that rote learning has “a lot to be said against it”, but added: “As the phrase ‘learning by heart’ implies, this sort of remembering places the poem at a central place in ourselves and makes it precious.”
          <Text>
            When Thatcher joined the campaign trail for the 2001 general election...
          </Text>
        </Text>
      </View>
    </View>
    <View>
      <View>
        <Video
          accountId="5436121857001"
          height={421.875}
          policyKey="BCpkADawqM0NK0Rq8n6sEQyWykemrqeSmIQqqVt3XBrdpl8TYlvqN3hwKphBJRnkPgx6WAbozCW_VgTOBCNf1AQRh8KnmXSXfveQalRc5-pyNlSod5XzP99If2U"
          poster={
            Object {
              "uri": "https://www.thetimes.co.uk/imageserver/image/%2Fmethode%2Ftimes%2Fprod%2Fweb%2Fbin%2F0c0309d4-1aeb-11e8-9010-1eef6ba5d3de.jpg?crop=1024%2C576%2C0%2C0",
            }
          }
          videoId="5831024132001"
          width={750}
        />
        <View>
          <View>
            <Text>
              This is video caption
            </Text>
          </View>
        </View>
      </View>
    </View>
    <View>
      <View>
        <Text
          selectable={true}
        >
          <Text>
            Baroness Thatcher, LG, OM, PC, FRS, Prime Minister, 1979-90, was born on October 13, 1925. She died on April 8, 2013, aged 87.
          </Text>
        </Text>
      </View>
    </View>
    <View>
      <View>
        <Video
          accountId="5436121857001"
          height={421.875}
          policyKey="BCpkADawqM0NK0Rq8n6sEQyWykemrqeSmIQqqVt3XBrdpl8TYlvqN3hwKphBJRnkPgx6WAbozCW_VgTOBCNf1AQRh8KnmXSXfveQalRc5-pyNlSod5XzP99If2U"
          poster={
            Object {
              "uri": "https://www.thetimes.co.uk/imageserver/image/%2Fmethode%2Ftimes%2Fprod%2Fweb%2Fbin%2F0c0309d4-1aeb-11e8-9010-1eef6ba5d3de.jpg?crop=1024%2C576%2C0%2C0",
            }
          }
          videoId="5831024132001"
          width={750}
        />
        <View>
          <View>
            <Text>
              This is video caption secondary
            </Text>
          </View>
        </View>
      </View>
    </View>
    <View>
      <View>
        <ArticleImage
          captionOptions={
            Object {
              "caption": "All the latest stories in culture and books.",
              "credits": "The credits",
            }
          }
          imageOptions={
            Object {
              "display": "primary",
              "ratio": "1500:1000",
              "uri": "https://www.thetimes.co.uk/imageserver/image/methode%2Ftimes%2Fprod%2Fweb%2Fbin%2Fb7ac990a-d872-11e7-9b16-edd3826708d6.jpg?crop=3959%2C2639%2C0%2C0",
            }
          }
        />
      </View>
    </View>
    <View>
      <View>
        <Text
          selectable={true}
        >
          When Thatcher joined the campaign trail for the 2001 general election, she was followed closely by the media. After several minor strokes, however, it was announced in 2002 that she would cut back her heavy schedule. 
        </Text>
      </View>
    </View>
    <View>
      <View>
        <View>
          <View>
            <RCTWebView
              injectedJavaScript="window.reactBridgePostMessage = window.postMessage; window.postMessage = String(Object.hasOwnProperty).replace('hasOwnProperty', 'postMessage');"
              messagingEnabled={true}
              scalesPageToFit={true}
              source={
                Object {
                  "uri": "https://cwfiyvo20d.execute-api.eu-west-1.amazonaws.com/dev/component/d2f83305-d558-4f78-f582-32115c659355",
                }
              }
            />
          </View>
        </View>
      </View>
    </View>
    <View>
      <View>
        <Text
          selectable={true}
        >
          Thatcher plainly admired the style and determination of Tony Blair, who crushed John Major’s Conservatives in the 1997 election, and he for his part seemed content from time to time to be compared with her as a strong leader. She regularly revisited some of the issues that had dominated her years in office; for example she gave strong support to the development of democracy and the protection of the rule of law in Hong Kong as promised in the treaty she had signed with China. In 1999, in her first conference speech since her resignation as leader, she defended General Augusto Pinochet, the former dictator of Chile, who was held in Britain after his arrest in 1998. 
        </Text>
      </View>
    </View>
    <View>
      <View>
        <ArticleImage
          captionOptions={
            Object {
              "caption": "All the latest stories in culture and books.",
              "credits": "The credits",
            }
          }
          imageOptions={
            Object {
              "display": "secondary",
              "ratio": "3:2",
              "uri": "https://www.thetimes.co.uk/imageserver/image/methode%2Ftimes%2Fprod%2Fweb%2Fbin%2F4476fabc-be54-11e7-b58a-4186f6049f2e.jpg?crop=5760%2C3840%2C0%2C0",
            }
          }
        />
      </View>
    </View>
    <View />
  </View>
</RCTScrollView>
`;

exports[`3. an article with a video as the lead asset 1`] = `
<RCTScrollView
  initialNumToRender={10}
  maxToRenderPerBatch={10}
  numColumns={1}
  onEndReachedThreshold={2}
  scrollEventThrottle={50}
  updateCellsBatchingPeriod={50}
  windowSize={21}
>
  <View>
    <View>
      <View>
        <Video
          accountId="5436121857001"
          height={421.875}
          policyKey="1.2.3.4"
          poster={
            Object {
              "uri": "https://video169.io",
            }
          }
          videoId="5831024132001"
          width={750}
        />
      </View>
    </View>
    <View>
      <View>
        <View>
          <VideoLabel
            color="#333333"
            title={null}
          />
        </View>
        <Text
          selectable={true}
        >
          Some Headline
        </Text>
      </View>
    </View>
    <View>
      <View>
        <View>
          <Text>
            Friday March 13 2015, 6:54pm, The Times
          </Text>
        </View>
      </View>
    </View>
    <View>
      <View>
        <Text
          selectable={true}
        >
          Some content
        </Text>
      </View>
    </View>
    <View />
    <View>
      <ArticleComments
        articleId="198c4b2f-ecec-4f34-be53-c89f83bc1b44"
        url="https://url.io"
      />
    </View>
  </View>
</RCTScrollView>
`;

exports[`4. an article with no lead asset 1`] = `
<RCTScrollView
  initialNumToRender={10}
  maxToRenderPerBatch={10}
  numColumns={1}
  onEndReachedThreshold={2}
  scrollEventThrottle={50}
  updateCellsBatchingPeriod={50}
  windowSize={21}
>
  <View>
    <View>
      <View>
        <Text
          selectable={true}
        >
          Some Headline
        </Text>
      </View>
    </View>
    <View>
      <View>
        <View>
          <Text>
            Friday March 13 2015, 6:54pm, The Times
          </Text>
        </View>
      </View>
    </View>
    <View>
      <View>
        <Text
          selectable={true}
        >
          Some content
        </Text>
      </View>
    </View>
    <View />
    <View>
      <ArticleComments
        articleId="198c4b2f-ecec-4f34-be53-c89f83bc1b44"
        url="https://url.io"
      />
    </View>
  </View>
</RCTScrollView>
`;

exports[`5. an article with no headline falls back to use shortheadline 1`] = `
<RCTScrollView
  initialNumToRender={10}
  maxToRenderPerBatch={10}
  numColumns={1}
  onEndReachedThreshold={2}
  scrollEventThrottle={50}
  updateCellsBatchingPeriod={50}
  windowSize={21}
>
  <View>
    <View>
      <View>
        <Text
          selectable={true}
        >
          Some Short Headline
        </Text>
      </View>
    </View>
    <View>
      <View>
        <View>
          <Text>
            Friday March 13 2015, 6:54pm, The Times
          </Text>
        </View>
      </View>
    </View>
    <View>
      <View>
        <Text
          selectable={true}
        >
          Some content
        </Text>
      </View>
    </View>
    <View />
    <View>
      <ArticleComments
        articleId="198c4b2f-ecec-4f34-be53-c89f83bc1b44"
        url="https://url.io"
      />
    </View>
  </View>
</RCTScrollView>
`;

exports[`6. an article with ads 1`] = `
<RCTScrollView
  initialNumToRender={10}
  maxToRenderPerBatch={10}
  numColumns={1}
  onEndReachedThreshold={2}
  scrollEventThrottle={50}
  updateCellsBatchingPeriod={50}
  windowSize={21}
>
  <View>
    <View>
      <View>
        <Text
          selectable={true}
        >
          Some Headline
        </Text>
      </View>
    </View>
    <View>
      <View>
        <View>
          <Text>
            Friday March 13 2015, 6:54pm, The Times
          </Text>
        </View>
      </View>
    </View>
    <View>
      <Ad
        contextUrl="https://url.io"
        section="Some Section"
        slotName="native-inline-ad"
      />
    </View>
    <View />
    <View>
      <ArticleComments
        articleId="198c4b2f-ecec-4f34-be53-c89f83bc1b44"
        url="https://url.io"
      />
    </View>
  </View>
</RCTScrollView>
`;

exports[`7. an article loading state 1`] = `
<View>
  <ActivityIndicator
    animating={true}
    color="#999999"
    hidesWhenStopped={true}
    size="large"
  />
</View>
`;

exports[`8. an article with updated byline 1`] = `
<RCTScrollView
  initialNumToRender={10}
  maxToRenderPerBatch={10}
  numColumns={1}
  onEndReachedThreshold={2}
  scrollEventThrottle={50}
  updateCellsBatchingPeriod={50}
  windowSize={21}
>
  <View>
    <View>
      <View>
        <Text
          selectable={true}
        >
          Some Headline
        </Text>
      </View>
    </View>
    <View>
      <View>
        <View>
          <Text>
            Friday March 13 2015, 6:54pm, The Times
          </Text>
        </View>
      </View>
    </View>
    <View>
      <View>
        <Text
          selectable={true}
        >
          Some content
        </Text>
      </View>
    </View>
    <View />
    <View>
      <ArticleComments
        articleId="198c4b2f-ecec-4f34-be53-c89f83bc1b44"
        url="https://url.io"
      />
    </View>
  </View>
</RCTScrollView>
`;

exports[`8. an article with updated byline 2`] = `
<RCTScrollView
  initialNumToRender={10}
  maxToRenderPerBatch={10}
  numColumns={1}
  onEndReachedThreshold={2}
  scrollEventThrottle={50}
  updateCellsBatchingPeriod={50}
  windowSize={21}
>
  <View>
    <View>
      <View>
        <Text
          selectable={true}
        >
          Some Headline
        </Text>
      </View>
    </View>
    <View>
      <View>
        <View>
          <Text>
            <ArticleBylineWithLinks
              ast={
                Array [
                  Object {
                    "attributes": Object {},
                    "children": Array [
                      Object {
                        "attributes": Object {
                          "value": "Some byline",
                        },
                        "children": Array [],
                        "name": "text",
                      },
                    ],
                    "name": "inline",
                  },
                ]
              }
              color="#FFFFFF"
            />
          </Text>
        </View>
        <View>
          <Text>
            Friday March 13 2015, 6:54pm, The Times
          </Text>
        </View>
      </View>
    </View>
    <View>
      <View>
        <Text
          selectable={true}
        >
          Some content
        </Text>
      </View>
    </View>
    <View />
    <View>
      <ArticleComments
        articleId="198c4b2f-ecec-4f34-be53-c89f83bc1b44"
        url="https://url.io"
      />
    </View>
  </View>
</RCTScrollView>
`;

exports[`9. an article that moves to a loading state 1`] = `
<RCTScrollView
  initialNumToRender={10}
  maxToRenderPerBatch={10}
  numColumns={1}
  onEndReachedThreshold={2}
  scrollEventThrottle={50}
  updateCellsBatchingPeriod={50}
  windowSize={21}
>
  <View>
    <View>
      <View>
        <Text
          selectable={true}
        >
          Some Headline
        </Text>
      </View>
    </View>
    <View>
      <View>
        <View>
          <Text>
            Friday March 13 2015, 6:54pm, The Times
          </Text>
        </View>
      </View>
    </View>
    <View>
      <View>
        <Text
          selectable={true}
        >
          Some content
        </Text>
      </View>
    </View>
    <View />
    <View>
      <ArticleComments
        articleId="198c4b2f-ecec-4f34-be53-c89f83bc1b44"
        url="https://url.io"
      />
    </View>
  </View>
</RCTScrollView>
`;

exports[`9. an article that moves to a loading state 2`] = `
<View>
  <ActivityIndicator
    animating={true}
    color="#999999"
    hidesWhenStopped={true}
    size="large"
  />
</View>
`;
